tests/integration.rs

```diff
@@ -3324,6 +3324,64 @@ mod tests {
         });
     }
 
+    #[cfg_attr(not(feature = "mmio"), test)]
+    fn test_virtio_mem() {
+        test_block!(tb, "", {
+            let mut clear = ClearDiskConfig::new();
+            let guest = Guest::new(&mut clear);
+            let api_socket = temp_api_path(&guest.tmp_dir);
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+            let mut kernel_path = workload_path;
+            kernel_path.push("vmlinux");
+
+            let mut child = GuestCommand::new(&guest)
+                .args(&["--cpus", "boot=2,max=4"])
+                .args(&["--memory", "size=512M,hotplug_method=virtio-mem,hotplug_size=8192M"])
+                .args(&["--kernel", kernel_path.to_str().unwrap()])
+                .args(&["--cmdline", "root=PARTUUID=6fb4d1a8-6c8c-4dd7-9f7c-1fe0b9f2574c console=tty0 console=ttyS0,115200n8 console=hvc0 quiet init=/usr/lib/systemd/systemd-bootchart initcall_debug tsc=reliable no_timer_check noreplace-smp cryptomgr.notests rootfstype=ext4,btrfs,xfs kvm-intel.nested=1 rw"])
+                .default_disks()
+                .default_net()
+                .args(&["--api-socket", &api_socket])
+                .spawn()
+                .unwrap();
+
+            thread::sleep(std::time::Duration::new(20, 0));
+
+            aver!(tb, guest.get_total_memory().unwrap_or_default() > 491_000);
+
+            guest
+                .ssh_command("echo online | sudo tee /sys/devices/system/memory/auto_online_blocks")
+                .unwrap_or_default();
+
+            // Add RAM to the VM
+            let desired_ram = 1024 << 20;
+            resize_command(&api_socket, None, Some(desired_ram));
+
+            thread::sleep(std::time::Duration::new(10, 0));
+            aver!(tb, guest.get_total_memory().unwrap_or_default() > 982_000);
+
+            // Add RAM to the VM
+            let desired_ram = 2048 << 20;
+            resize_command(&api_socket, None, Some(desired_ram));
+
+            thread::sleep(std::time::Duration::new(10, 0));
+            aver!(tb, guest.get_total_memory().unwrap_or_default() > 1_964_000);
+
+            // Remove RAM to the VM
+            let desired_ram = 1024 << 20;
+            resize_command(&api_socket, None, Some(desired_ram));
```

Comment:
This one will work because we're not setting `current_ram` from the virtio-mem code path when up-sizing. Then we you down-size back to the original setting, `memory_manager.resize()` works because we check that `if desired_ram >= self.current_ram`.
But I think this is incorrect, `memory_manager.current_ram` should reflect that actual amount of memory we added to the guest, be it through ACPI or virtio-mem.

Reply:
Even if current_ram is not update, the code is still work because it just check guest.get_total_memory() that get date from the guest memory.
```
        fn get_total_memory(&self) -> Result<u32, Error> {
            Ok(self
                .ssh_command("grep MemTotal /proc/meminfo | grep -o \"[0-9]*\"")?
                .trim()
                .parse()
                .map_err(Error::Parsing)?)
        }
```